OMCompiler/Compiler/BackEnd/BackendDAEUtil.mo

```diff
@@ -4390,31 +4390,36 @@ protected
   BackendDAE.Equation eq;
   BackendDAE.VarKind varKind;
   DAE.ComponentRef cr;
+  String eqnstr;
   constant Boolean debug = false;
 algorithm
   // Get equation and variable from integer index
   try
     BackendDAE.EQSYSTEM(orderedVars=vars, orderedEqs=eqns) := isyst;
     BackendDAE.SHARED(globalKnownVars=globalKnownVars) := ishared;
+
+    eq := BackendEquation.get(eqns, inEquation);
+    BackendDAE.VAR(varName = cr , varKind = varKind) := BackendVariable.getVarAt(vars, inVariable);
+    // Compute solvability
+    solvability := match(varKind)
+      // Case variable is a state
+      case(BackendDAE.STATE()) algorithm
+          v := BackendVariable.createAliasDerVar(cr);
+          cr := BackendVariable.varCref(v);
+        then testSolvabilityForEquation(cr, eq, varArray, vars, globalKnownVars);
+      else
+        then testSolvabilityForEquation(cr, eq, varArray, vars, globalKnownVars);
+    end match;
+
+    if debug then
+      print("\n solvability: " + boolString(solvability) + " \n");
+    end if;
+
   else
-    //Todo: Error
+    eqnstr := stringAppendList({"findSolvabelVarInEquation failed\n Could not get equation and variable from integer index."});
```

Comment:
While you are at it, could you change the name to `findSolvableVarInEquation `?

Reply:
Will be removed any way